pkg/scheduler/framework/v1alpha1/framework_test.go

```diff
@@ -319,25 +379,37 @@ func buildConfigWithWeights(weights map[string]int32, ps ...string) *config.Plug
 }
 
 type injectedResult struct {
-	scoreRes     int
-	normalizeRes int
-	scoreErr     bool
-	normalizeErr bool
+	ScoreRes     int  `json:"scoreRes,omitempty"`
+	NormalizeRes int  `json:"normalizeRes,omitempty"`
+	ScoreErr     bool `json:"scoreErr,omitempty"`
+	NormalizeErr bool `json:"normalizeErr,omitempty"`
 }
 
 func setScoreRes(inj injectedResult) (int, *Status) {
-	if inj.scoreErr {
+	if inj.ScoreErr {
 		return 0, NewStatus(Error, "injecting failure.")
 	}
-	return inj.scoreRes, nil
+	return inj.ScoreRes, nil
 }
 
 func injectNormalizeRes(inj injectedResult, scores NodeScoreList) *Status {
-	if inj.normalizeErr {
+	if inj.NormalizeErr {
 		return NewStatus(Error, "injecting failure.")
 	}
 	for i := range scores {
-		scores[i].Score = inj.normalizeRes
+		scores[i].Score = inj.NormalizeRes
 	}
 	return nil
 }
+
+func decodeInjectedResult(injUnknown *runtime.Unknown) (*injectedResult, error) {
```

Comment:
we can use DecodeInto (exist in registry.go) instead of defining a new function for decoding.